src/Storages/ObjectStorage/DataLakes/IcebergMetadata.cpp

```diff
@@ -403,23 +401,67 @@ DataLakeMetadataPtr IcebergMetadata::create(
     const Poco::JSON::Object::Ptr & object = json.extract<Poco::JSON::Object::Ptr>();
 
     auto format_version = object->getValue<int>("format-version");
-    auto [schema, schema_id]
-        = parseTableSchema(object, format_version, local_context->getSettingsRef()[Setting::iceberg_engine_ignore_schema_evolution]);
-
-    auto current_snapshot_id = object->getValue<Int64>("current-snapshot-id");
     auto snapshots = object->get("snapshots").extract<Poco::JSON::Array::Ptr>();
 
     String manifest_list_file;
-    for (size_t i = 0; i < snapshots->size(); ++i)
+    Int32 schema_id = -1;
+
+    // Use the current snapshot (by default) or find the appropriate snapshot based on timestamp
+    Int64 query_timestamp = local_context->getSettingsRef()[Setting::iceberg_query_at_timestamp_ms];
+    if (query_timestamp == 0)
     {
-        const auto snapshot = snapshots->getObject(static_cast<UInt32>(i));
-        if (snapshot->getValue<Int64>("snapshot-id") == current_snapshot_id)
+        // Use current snapshot
+        auto current_snapshot_id = object->getValue<Int64>("current-snapshot-id");
+        for (size_t i = 0; i < snapshots->size(); ++i)
+        {
+            const auto snapshot = snapshots->getObject(static_cast<UInt32>(i));
+            if (snapshot->getValue<Int64>("snapshot-id") == current_snapshot_id)
+            {
+                const auto path = snapshot->getValue<String>("manifest-list");
+                manifest_list_file = std::filesystem::path(configuration->getPath()) / "metadata" / std::filesystem::path(path).filename();
+                schema_id = snapshot->getValue<Int32>("schema-id");
+                break;
+            }
+        }
+
```

Comment:
Could you, please, add chassert (CH version of assert) that schema_id == current_schema_id = metadata_object->getValue<int>("current-schema-id")), please? Or you can throw a LOGICAL_ERROR in case it is not